app/src/androidTest/java/org/simple/clinic/user/UserSessionAndroidTest.kt

```diff
@@ -60,4 +74,65 @@ class UserSessionAndroidTest {
     val (accessToken) = userSession.accessToken()
     assertThat(accessToken).isNull()
   }
+
+  @Test
+  fun when_logging_in_from_registration_entry_user_should_be_logged_in_locally() {
+    val facilities = facilityApi.pull(10)
+        .map { it.facilities }
+        .map { it.map { it.toDatabaseModel(SyncStatus.DONE) } }
+        .blockingGet()
+    appDatabase.facilityDao().save(facilities)
+
+    val selectedFacilities = facilities.subList(0, 2)
+    val ongoingRegistrationEntry = patientFaker.ongoingRegistrationEntry(selectedFacilities)
+    userSession.saveOngoingRegistrationEntry(ongoingRegistrationEntry)
+        .andThen(userSession.loginFromOngoingRegistrationEntry())
+        .blockingAwait()
+
+    assertThat(userSession.isUserLoggedIn()).isTrue()
+    val (loggedInUser) = userSession.loggedInUser().blockingFirst()
+    assertThat(loggedInUser!!.status).isEqualTo(UserStatus.WAITING_FOR_APPROVAL)
+
+    val currentFacility = facilityRepository
+        .currentFacility(userSession)
+        .blockingFirst()
+    assertThat(currentFacility.uuid).isEqualTo(selectedFacilities.first().uuid)
+
+    val postLoginRegistrationEntry = userSession.ongoingRegistrationEntry().blockingGet()
+    assertThat(postLoginRegistrationEntry).isEqualTo(OngoingRegistrationEntry())
+  }
+
+  @Test
+  fun when_user_is_logged_out_then_all_app_data_should_get_cleared() {
```

Comment:
Should we also check if shared preferences also get cleared?

Reply:
Oh thanks for catching this. I forgot to add shared preferences.